Link typo in drivers.rst

dhalbert opened the pull request.

kattni commented:
Travis is failing to build due to an issue with one of the build's download links. Dan will put a fix in eventually. Until then, this PR is approved but waiting on the Travis fix.

dhalbert commented:
This minor PR is now fixed and re-purposed for a 3.1.2 release. I fixed the nRF 52832 build and removed the 52840 build, which depended on an alpha version of the SD that is no longer available, and would require significant code changes to upgrade to a non-alpha version.

The main point of the 3.1.2 release is to include `serial_bytes_available`, which @ATMakersBIll is depending on for a Learn Guide.
